### ISSUE TITLE
Test suite seems incompatible w/ Django 1.11

### ISSUE THREAD

**Issue by toolness** [18F/cg-django-uaa#21]
The [development instructions](http://cg-django-uaa.readthedocs.io/en/latest/developing.html) actually appear to install the very latest version of Django, which is currently 1.11, under which `setup.py test` produces the following output:

```pytb
Traceback (most recent call last):
  File "setup.py", line 118, in <module>
    'Topic :: Utilities'],
  File "/usr/lib/python3.6/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/lib/python3.6/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/home/atul/Documents/18f/cg-django-uaa/venv/lib/python3.6/site-packages/setuptools/command/test.py", line 211, in run
    self.run_tests()
  File "/home/atul/Documents/18f/cg-django-uaa/venv/lib/python3.6/site-packages/setuptools/command/test.py", line 234, in run_tests
    **exit_kwarg
  File "/usr/lib/python3.6/unittest/main.py", line 93, in __init__
    self.parseArgs(argv)
  File "/usr/lib/python3.6/unittest/main.py", line 140, in parseArgs
    self.createTests()
  File "/usr/lib/python3.6/unittest/main.py", line 147, in createTests
    self.module)
  File "/usr/lib/python3.6/unittest/loader.py", line 219, in loadTestsFromNames
    suites = [self.loadTestsFromName(name, module) for name in names]
  File "/usr/lib/python3.6/unittest/loader.py", line 219, in <listcomp>
    suites = [self.loadTestsFromName(name, module) for name in names]
  File "/usr/lib/python3.6/unittest/loader.py", line 204, in loadTestsFromName
    test = obj()
  File "/home/atul/Documents/18f/cg-django-uaa/uaa_client/runtests.py", line 74, in run_tests
    failures = test_runner.run_tests(['uaa_client.tests'])
  File "/home/atul/Documents/18f/cg-django-uaa/venv/lib/python3.6/site-packages/django/test/runner.py", line 602, in run_tests
    self.run_checks()
  File "/home/atul/Documents/18f/cg-django-uaa/venv/lib/python3.6/site-packages/django/test/runner.py", line 562, in run_checks
    call_command('check', verbosity=self.verbosity)
  File "/home/atul/Documents/18f/cg-django-uaa/venv/lib/python3.6/site-packages/django/core/management/__init__.py", line 130, in call_command
    return command.execute(*args, **defaults)
  File "/home/atul/Documents/18f/cg-django-uaa/venv/lib/python3.6/site-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/home/atul/Documents/18f/cg-django-uaa/venv/lib/python3.6/site-packages/django/core/management/commands/check.py", line 68, in handle
    fail_level=getattr(checks, options['fail_level']),
  File "/home/atul/Documents/18f/cg-django-uaa/venv/lib/python3.6/site-packages/django/core/management/base.py", line 359, in check
    include_deployment_checks=include_deployment_checks,
  File "/home/atul/Documents/18f/cg-django-uaa/venv/lib/python3.6/site-packages/django/core/management/base.py", line 346, in _run_checks
    return checks.run_checks(**kwargs)
  File "/home/atul/Documents/18f/cg-django-uaa/venv/lib/python3.6/site-packages/django/core/checks/registry.py", line 81, in run_checks
    new_errors = check(app_configs=app_configs)
  File "/home/atul/Documents/18f/cg-django-uaa/venv/lib/python3.6/site-packages/django/core/checks/urls.py", line 16, in check_url_config
    return check_resolver(resolver)
  File "/home/atul/Documents/18f/cg-django-uaa/venv/lib/python3.6/site-packages/django/core/checks/urls.py", line 26, in check_resolver
    return check_method()
  File "/home/atul/Documents/18f/cg-django-uaa/venv/lib/python3.6/site-packages/django/urls/resolvers.py", line 254, in check
    for pattern in self.url_patterns:
  File "/home/atul/Documents/18f/cg-django-uaa/venv/lib/python3.6/site-packages/django/utils/functional.py", line 35, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/home/atul/Documents/18f/cg-django-uaa/venv/lib/python3.6/site-packages/django/urls/resolvers.py", line 405, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
  File "/home/atul/Documents/18f/cg-django-uaa/venv/lib/python3.6/site-packages/django/utils/functional.py", line 35, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/home/atul/Documents/18f/cg-django-uaa/venv/lib/python3.6/site-packages/django/urls/resolvers.py", line 398, in urlconf_module
    return import_module(self.urlconf_name)
  File "/home/atul/Documents/18f/cg-django-uaa/venv/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 978, in _gcd_import
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load
  File "<frozen importlib._bootstrap>", line 950, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 655, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 205, in _call_with_frames_removed
  File "/home/atul/Documents/18f/cg-django-uaa/uaa_client/tests/urls.py", line 10, in <module>
    url(r'^auth/', include('uaa_client.urls', **_kwargs)),
  File "/home/atul/Documents/18f/cg-django-uaa/venv/lib/python3.6/site-packages/django/conf/urls/__init__.py", line 50, in include
    urlconf_module = import_module(urlconf_module)
  File "/home/atul/Documents/18f/cg-django-uaa/venv/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 978, in _gcd_import
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load
  File "<frozen importlib._bootstrap>", line 950, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 655, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 205, in _call_with_frames_removed
  File "/home/atul/Documents/18f/cg-django-uaa/uaa_client/urls.py", line 18, in <module>
    validate_configuration()
  File "/home/atul/Documents/18f/cg-django-uaa/uaa_client/configuration.py", line 17, in validate_configuration
    'non-empty string'.format(setting)
django.core.exceptions.ImproperlyConfigured: settings.UAA_CLIENT_ID must be defined as a non-empty string
```

**Comment by toolness** [comment]
I think this is due to the following item from the [Django 1.11 release notes](https://docs.djangoproject.com/en/1.11/releases/1.11/#tests):

> `DiscoverRunner` now runs the system checks at the start of a test run. Override the `DiscoverRunner.run_checks()` method if you want to disable that.

Because those system checks run at the start of a test run, #18 gets triggered early. Yuck.

**Comment by toolness** [comment]
Fixed by #34!
